src/app/test_executive/test_executive.ml

```diff
@@ -80,53 +126,10 @@ let main inputs =
   let net_manager_ref : Engine.Network_manager.t option ref = ref None in
   let log_engine_ref : Engine.Log_engine.t option ref = ref None in
   let cleanup_deferred_ref = ref None in
-  let dispatch_cleanup reason (test_result : unit Malleable_error.t) :
-      unit Deferred.t =
-    let cleanup () : unit Deferred.t =
-      let%bind () =
-        Option.value_map !net_manager_ref ~default:Deferred.unit
-          ~f:Engine.Network_manager.cleanup
-      in
-      let log_engine_cleanup_result =
-        Option.value_map !log_engine_ref
-          ~default:(Malleable_error.return Test_error.Set.empty)
-          ~f:Engine.Log_engine.destroy
-      in
-      let%bind log_engine_error_set =
-        match%map Malleable_error.lift_error_set log_engine_cleanup_result with
-        | Ok (remote_error_set, internal_error_set) ->
-            Test_error.Set.combine [remote_error_set; internal_error_set]
-        | Error internal_error_set ->
-            internal_error_set
-      in
-      let%bind test_error_set =
-        match%map Malleable_error.lift_error_set test_result with
-        | Ok ((), error_set) ->
-            error_set
-        | Error error_set ->
-            error_set
-      in
-      report_test_errors
-        (Test_error.Set.combine [test_error_set; log_engine_error_set])
-    in
-    let%bind test_error_str =
-      Malleable_error.hard_error_to_string test_result
-    in
-    match !cleanup_deferred_ref with
-    | Some deferred ->
-        [%log error]
-          "additional call to cleanup testnet while already cleaning up \
-           ($reason, $error)"
-          ~metadata:
-            [("reason", `String reason); ("error", `String test_error_str)] ;
-        deferred
-    | None ->
-        [%log info] "cleaning up testnet ($reason, $error)"
-          ~metadata:
-            [("reason", `String reason); ("error", `String test_error_str)] ;
-        let deferred = cleanup () in
-        cleanup_deferred_ref := Some deferred ;
-        deferred
+  let dispatch_cleanup_PARTIAL =
```

Comment:
nit: rename `dispatch_cleanup_PARTIAL` to `f_dispatch_cleanup` or just `cleanup` or something else? `PARTIAL` to me suggests that the cleanup itself is partial